DSharpPlus.Commands/Invocation/CommandEmitUtil.cs

```diff
@@ -0,0 +1,142 @@
+namespace DSharpPlus.Commands.Invocation;
+
+using System;
+using System.Reflection;
+using System.Reflection.Emit;
+using System.Threading.Tasks;
+
+/// <summary>
+/// Contains utilities to conveniently await all (supported) commands as ValueTasks.
+/// </summary>
+internal static class CommandEmitUtil
+{
+    /// <summary>
+    /// Creates a wrapper function to invoke a command.
+    /// </summary>
+    /// <param name="method">The corresponding MethodInfo for this command.</param>
+    /// <exception cref="InvalidOperationException">Thrown if the command returns anything but ValueTask and Task.</exception>
+    public static Func<object?, object?[], ValueTask> GetCommandInvocationFunc(MethodInfo method)
+    {
+        if (method.ReturnType == typeof(ValueTask))
+        {
+            return GetValueTaskFunc(method);
+        }
+
+        else if (method.ReturnType == typeof(Task))
+        {
+            return GetTaskFunc(method);
+        }
+
+        throw new InvalidOperationException
```

Comment:
Does this mean a method's return type *must*  be a `Task` or `ValueTask`? If so, the command builder's `From` method should be updated to check this.

Reply:
i don't think they should, other executors are should be free to use other return types imo